### PR TITLE
gawk, chroot-gawk: update to 5.2.2.

### PR DESCRIPTION
**Pull request by mhmdanas**
#### Testing the changes
- I tested the changes in this PR: **YES**

### CODE REVIEW

File: srcpkgs/gawk/template [hunk 1]
```diff
@@ -10,8 +10,10 @@ maintainer="Enno Boland <gottox@voidlinux.org>"
 license="GPL-3.0-or-later"
 homepage="https://directory.fsf.org/wiki/Gawk"
 distfiles="${GNU_SITE}/${pkgname}/${pkgname}-${version}.tar.xz"
-checksum=d87629386e894bbea11a5e00515fc909dc9b7249529dad9e6a3a2c77085f7ea2
+checksum=673553b91f9e18cc5792ed51075df8d510c9040f550a6f74e09c9add243a7e4f
 lib32disabled=yes
+# PIE is disabled for persistent malloc.
+nopie=yes
```
[the remark this answers is not in the document]
Comment (sgn) [reply srcpkgs/gawk/template:16]: Please disable it.

[the remark this answers is not in the document]
Comment (sgn) [reply srcpkgs/gawk/template:16]: `mmap(2)` and `munmap(2)` are POSIX, Linux supports them.

**Comment by classabbyamp** [comment]
let's hold off on this for now - q66 has found some issues with stack overflows in 5.2 even with pma disabled

**Comment by tornaria** [comment]
changelog at https://git.savannah.gnu.org/cgit/gawk.git/plain/NEWS

**Pull request closed without merging**